File: FE/src/models/thuocModel.js
import { formatNumber } from "./formatNumberModel";
const toJson = (item) => {
    //console.log("LOG DON GIA : ",  formatNumber.FormatNumber(item.donGia));
    return {
        id: item.id,
        code: item.code,
        name: item.name,
        donViTinhs : item.donViTinhs,
        donGia :  formatNumber.FormatNumber(item.donGia),
        thueGTGT  : item.thueGTGT,
        phanTramChieuThue : item.phanTramChieuThue,
        chiecKhau : item.chiecKhau,
        trangThai : item.trangThai,
        loaiDichVuYTe : item.loaiDichVuYTe,
        giaMuaVao : formatNumber.FormatNumber(item.giaMuaVao),
        giaLamTron: formatNumber.FormatNumber(item.giaLamTron),
        phiTuVan : formatNumber.FormatNumber(item.phiTuVan),
        phiTiem : formatNumber.FormatNumber(item.phiTiem),
        donGiaBanRa:formatNumber.FormatNumber(item.donGiaBanRa),
    }
}



const fromJson = (item) => {
    return {
        id: item.id,
        code: item.code,
        name: item.name,
        donViTinhs : item.donViTinhs,
        donGia : item.donGia,
        thueGTGT  : item.thueGTGT,
        phanTramChieuThue : item.phanTramChieuThue,
        chiecKhau : item.chiecKhau,
        trangThai : item.trangThai,
        loaiDichVuYTe : item.loaiDichVuYTe,
        giaMuaVao : item.giaMuaVao,
        giaLamTron: item.giaLamTron,
        phiTuVan : item.phiTuVan,
        phiTiem : item.phiTiem,
        donGiaBanRa:item.donGiaBanRa,
    }
}



const baseJson = () => {
    return {
        id: null,
        code: null,
        name: null,
        donViTinhs : null,
        donGia : null,
        thueGTGT  : null,
        phanTramChieuThue :  0,
        chiecKhau : 0,
        trangThai : null,
        loaiDichVuYTe :null,
        giaMuaVao : 0,
        giaLamTron: 0,
        phiTuVan : 0,
        phiTiem : 0,
        donGiaBanRa: 0
    }
}

const toListModel = (items) =>{
    if(items.length > 0){
        let data = [];
        items.map((value, index) =>{
            data.push(fromJson(value));
        })
        return data??[];
    }
    return [];
}
export const thuocModel = {
    toJson, fromJson, baseJson, toListModel
}

File: BE/HDDT_BE/Models/Professional/ThuocModel.cs
using HDDT_BE.Constants;
using HDDT_BE.Interfaces.Core;
using MongoDB.Bson.Serialization.Attributes;

namespace HDDT_BE.Models.Core;

public class ThuocModel : Audit, TEntity<string>
{
    
    public string Code { get; set; } // Code của hệ thống mới của mình 

    public int Sort { get; set; } = 0;
    
    public double DonGia { get; set; }
    
    public CommonModelShort ThueGTGT { get; set; }
    
    public int PhanTramChieuThue { get; set; }
    
    public int ChiecKhau { get; set; }
    
    public CommonModelShort TrangThai { get; set; }
    
    public double GiaMuaVao { get; set; }
    public string GiaMuaVaoShow 
    {
        get
        {
            return GiaMuaVao != null ? FormatMoney.ConvertToMoney(GiaMuaVao) : ""; 
        }
    }
    
    public double GiaLamTron { get; set; }
    public string GiaLamTronShow 
    {
        get
        {
            return GiaLamTron != null ? FormatMoney.ConvertToMoney(GiaLamTron) : ""; 
        }
    }
    
    public double PhiTuVan { get; set; }
    public string PhiTuVanShow 
    {
        get
        {
            return PhiTuVan != null ? FormatMoney.ConvertToMoney(PhiTuVan) : ""; 
        }
    }
    
    public double PhiTiem { get; set; }
    
    public string PhiTiemShow 
    {
        get
        {
            return PhiTiem != null ? FormatMoney.ConvertToMoney(PhiTiem) : ""; 
        }
    }
    
    public double DonGiaBanRa { get; set; }
    
    public string DonGiaBanRaShow 
    {
        get
        {
            return DonGiaBanRa != null ? FormatMoney.ConvertToMoney(DonGiaBanRa) : ""; 
        }
    }
    public List<CommonModelShort> DonViTinhs { get; set; }
    
    public CommonModelShort LoaiDichVuYTe { get; set; }
    
    public bool DuocGiamGia { get; set; } = false;

    public double SoTienGiamLanHai { get; set; }

}





public class ThuocModelShort
{
    [BsonId]
    [BsonRepresentation(MongoDB.Bson.BsonType.ObjectId)]
    public string Id { get; set; }

    public string Name { get; set; }
    
    public string UnsignedName { get; set; }
    
    public string Code { get; set; } // Code của hệ thống mới của mình
    
    public int Sort { get; set; }
    
    public double DonGia { get; set; }
    
    public CommonModelShort ThueGTGT { get; set; }
    
    public int PhanTramChieuThue { get; set; }
    
    public int ChiecKhau { get; set; }
    
    public CommonModelShort TrangThai { get; set; }
    
    public CommonModelShort LoaiHangHoa { get; set; }
    
    public double GiaMuaVao { get; set; }
    
    public double GiaLamTron { get; set; }
    
    public double PhiTuVan { get; set; }
    
    public double PhiTiem { get; set; }
    
    public double DonGiaBanRa { get; set; }
    
    public List<CommonModelShort> DonViTinhs { get; set; }
    public CommonModelShort LoaiDichVuYTe { get; set; }
    
    public bool DuocGiamGia { get; set; } = false;

    public double SoTienGiamLanHai { get; set; }
    
}